### ISSUE TITLE
Add support of dns resolution on inter-region vpc peering

### ISSUE THREAD

**Issue by tdegiacinto** [terraform-providers/terraform-provider-aws#6730]
Hi 
a few weeks ago AWS announced the support of dns resolution across multi-region vpc peering.
I tried to use it but I got the following error 
```
Error modifying VPC Peering Connection Options: OperationNotPermitted: Modifying VPC peering connection options AllowEgressFromLocalClassicLinkToRemoteVpc, AllowEgressFromLocalVpcToRemoteClassicLink is not supported for cross-region VPC peering connections
``` 
It seems that when we set `allow_remote_vpc_dns_resolution` option , the provider try to set the 3 available options which 2 of them are not supported.

### Terraform Version

terraform version : 0.11.10

### Affected Resource(s)

* aws_vpc_peering_connection
* aws_vpc_peering_connection_accepter
* aws_vpc_peering_connection_options

### Terraform Configuration Files

Here is my code 
```hcl
resource "aws_vpc_peering_connection_options" "options_requester" {
  vpc_peering_connection_id = "${aws_vpc_peering_connection.to_main_region.id}"

  requester {
    allow_remote_vpc_dns_resolution = true
  }
}
```

### Expected Behavior

Just change the dns resolution option

### Actual Behavior

Try to set all three options.

### Steps to Reproduce
Create an inter-region peering with dns resolution

**Comment by ewbankkit** [comment]
See also [here](https://github.com/terraform-providers/terraform-provider-aws/pull/3909#issuecomment-442290287).

**Comment by ewbankkit** [comment]
[This](https://aws.amazon.com/about-aws/whats-new/2018/11/announcing-support-for-dns-resolution-over-inter-region-vpc-peering/) is the announcement.

**Comment by tdegiacinto** [comment]
Oh nice. We can close this issue.

**Comment by ewbankkit** [comment]
@tdegiacinto Actually could you please reopen. The referenced PR did not address the cross-region case correctly and it would be nice to have a fresh issue to keep track. Thanks.

**Comment by imdhruva** [comment]
same issue here, getting the following error for enabling cross region DNS resolution:
```
Error: Error applying plan:

2 error(s) occurred:

* module.syd_peers_infra.aws_vpc_peering_connection_options.requester: 1 error(s) occurred:

* aws_vpc_peering_connection_options.requester: Error modifying VPC Peering Connection Options: OperationNotPermitted: Modifying VPC peering connection options AllowEgressFromLocalClassicLinkToRemoteVpc, AllowEgressFromLocalVpcToRemoteClassicLink is not supported for cross-region VPC peering connections
        status code: 400, request id: 2e6f012a-a2cf-45e0-a2ab-1d209f038f16
* module.syd_peers_infra.aws_vpc_peering_connection_options.destintation: 1 error(s) occurred:

* aws_vpc_peering_connection_options.destintation: Error modifying VPC Peering Connection Options: OperationNotPermitted: Modifying VPC peering connection options AllowEgressFromLocalClassicLinkToRemoteVpc, AllowEgressFromLocalVpcToRemoteClassicLink is not supported for cross-region VPC peering connections
        status code: 400, request id: 706cf983-37e3-484e-93a8-3f67af4e0cbf

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```
the connection options is inspired from the [example docs here](https://www.terraform.io/docs/providers/aws/r/vpc_peering_options.html) :
```
resource "aws_vpc_peering_connection_options" "requester" {
  vpc_peering_connection_id = "${aws_vpc_peering_connection.peer.id}"

  requester {
    allow_remote_vpc_dns_resolution = true
  }
}

resource "aws_vpc_peering_connection_options" "destintation" {
  provider                  = "aws.accepter"
  vpc_peering_connection_id = "${aws_vpc_peering_connection.peer.id}"

  accepter {
    allow_remote_vpc_dns_resolution = true
  }
}
```

Any estimated time of resolution for this one @ewbankkit ?

**Comment by ewbankkit** [comment]
I need to rebase the PR to fix merge conflicts.

**Comment by jalaziz** [comment]
We're facing this too. Hopefully the PR gets merged soon.

**Comment by tristan-weil** [comment]
I can confirm this PR fixes the problem

**Comment by tprakash17** [comment]
is this merged or still open?

**Comment by zelch** [comment]
This hasn't been merged yet, and the problem still exists.

**Comment by ewbankkit** [comment]
@tprakash17 @zelch Please add a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [this issue](https://github.com/terraform-providers/terraform-provider-aws/issues/6730#issue-387923296) if you haven't done so already.

**Comment by nickvelo** [comment]
this is a blocker for me. any updates on this?

**Comment by bgrabows1** [comment]
Any update?

**Comment by francis-edejer** [comment]
Sharing my temporary/dirty workaround for this issue till this issue gets released:

`resource "aws_vpc_peering_connection_accepter" "current-to-us-east-1" {
  vpc_peering_connection_id = aws_vpc_peering_connection.current-to-us-east-1[0].id
  auto_accept               = true

  tags     = var.default_tags
  provisioner "local-exec" {
    command = "aws --profile ${var.aws_profile} --region ${var.main_vpc_details["region"]} ec2 modify-vpc-peering-connection-options --vpc-peering-connection-id ${aws_vpc_peering_connection.current-to-us-east-1[0].id} --accepter-peering-connection-options AllowDnsResolutionFromRemoteVpc=true"
  }
  provisioner "local-exec" {
    command = "aws --profile ${var.aws_profile} --region ${var.vpc_details["region"]} ec2 modify-vpc-peering-connection-options --vpc-peering-connection-id ${aws_vpc_peering_connection.current-to-us-east-1[0].id} --requester-peering-connection-options AllowDnsResolutionFromRemoteVpc=true"
  }
  provider = aws.us_east
}`

Not ideal since it requires the CLI to be present and configured, but keeps every step of the provisioning process in terraform.

**Comment by completenovice** [comment]
Is there any news on this? It's a blocker for us too, we're wondering what needs to be done to get this fix released.

**Comment by bflad** [comment]
Support for this has been merged and will release with version 2.31.0 of the Terraform AWS Provider, next Thursday. Thanks to @ewbankkit for the implementation work.
